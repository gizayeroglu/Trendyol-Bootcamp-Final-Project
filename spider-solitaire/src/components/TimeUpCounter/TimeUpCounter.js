import React from 'react'
import { useState, useEffect } from 'react';

const TimeUpCounter = () => {
  const [ hours, setHours ] = useState(0);
  const [ minutes, setMinutes ] = useState(0);
  const [seconds, setSeconds ] =  useState(0);

  useEffect(()=>{
    let myInterval = setInterval(() => {
      if (seconds >= 0) {
        setSeconds(seconds + 1);
      }
      if(seconds === 59){
        setSeconds(0);
        setMinutes(minutes + 1);
      }
      if(seconds===59 && minutes===59){
        setSeconds(0);
        setMinutes(0);
        setHours(hours + 1);
      }

    }, 1000)

    return ()=> {
      clearInterval(myInterval);
    };
  });

  return (
    <>  
     Time: {hours < 10 ? `0${hours}` : hours}:{minutes < 10 ? `0${minutes}` : minutes}:{seconds < 10 ?  `0${seconds}` : seconds} 
    </>
  )
}

export default TimeUpCounter;