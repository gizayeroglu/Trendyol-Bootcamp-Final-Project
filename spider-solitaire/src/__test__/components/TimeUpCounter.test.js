import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

import TimeUpCounter from '../../components/TimeUpCounter/TimeUpCounter';

configure({ adapter: new Adapter() });

describe('Time Up Counter component test', function () {
  let wrapper;

  it('should render', () => {
    wrapper = shallow(<TimeUpCounter />);
    expect(wrapper).toMatchSnapshot();
  });
});