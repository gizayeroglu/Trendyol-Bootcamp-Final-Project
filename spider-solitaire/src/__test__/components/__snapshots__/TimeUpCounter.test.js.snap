// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Up Counter component test should render 1`] = `ShallowWrapper {}`;
